apps/docs/src/guide/contracts/transaction-dependency-estimation.md

```diff
@@ -0,0 +1,11 @@
+# Transaction Dependency Estimation
+
+Previously, we mentioned that a contract call might require you to manually specify external contracts or variable outputs.
+
+However, the SDK always automatically estimates these dependencies and double-checks if everything is in order whenever you invoke a contract function or attempt to send a transaction.
+
```

Comment:
@digorithm I believe that this behaviour here in the TS SDK is slightly different from the Rust SDK. The Rust SDK doesn't seem to automatically call `estimate_tx_dependencies` on each function invocation, while the TS SDK does that every single time automatically.

I feel like this would make the `estimateTxDependencies` function here in the TS SDK more of an _internal_ helper since SDK users will never have to call it manually.

Reply:
Yeah. I'd suggest that, in the near future, we should move to _manually_ calling `estimateTxDependencies`; the additional dry-run calls happening without the user's consent could be an undesired behavior. Or it could be an optional value we set at the call time. We can discuss that later.